pkg/build/api/types.go

```diff
@@ -425,6 +428,19 @@ type SourceBuildStrategy struct {
 	ForcePull bool
 }
 
+// JenkinsPipelineStrategy holds parameters specific to a Jenkins Pipeline build.
+type JenkinsPipelineBuildStrategy struct {
```

Comment:
A discussion today revealed some disagreements about naming.  It came down to whether we want to introduce in the future a "generic" pipeline type that deals only with pipeline templates and parameters, but doesn't allow end users to specify their own Jenkinsfiles.  This fits into other operational requirements we've heard - admins want to control the processes their teams use and define them centrally.

The suggestion was to call the type "Jenkinsfile" (to align with the process document that controls it).  This struct would be "JenkinsfileBuildStrategy".   An end user would select this strategy if they had or wanted to use a Jenkinsfile directly.  The reason to suggest Jenkinsfile over JenkinsPipeline is that the key element here is the Jenkinsfile (the pipeline plugin is just what processes it).  We said that if were to name the strategies over, we'd probably call the Docker type "Dockerfile" since that's the key document that defines the process.

The other (future) strategy would be called "Pipeline", and would be focused on those template types (like "approval-for-production" or "standard-microservice-team") which would heavily deal with parameterization and controlling what users could do.

Thoughts / feedback from folks?